File: Competetive programming/C++/Unit 5/Pairsumonious_Numbers.cpp
#include <iostream>
using namespace std;

int main() {

    int s1, s2, s3;
    int a, b, c;

    cout << "===== PAIRSUMONIOUS NUMBERS =====\n\n";

    cout << "Enter three pair sums:\n";
    cout << "Format:\n";
    cout << "a+b  a+c  b+c\n\n";

    cout << "Example:\n";
    cout << "3 4 5\n\n";

    cin >> s1 >> s2 >> s3;

    a = (s1 + s2 - s3) / 2;
    b = s1 - a;
    c = s2 - a;

    cout << "\nOriginal Numbers:\n";
    cout << a << " " << b << " " << c << endl;

    return 0;
}

File: Competetive programming/Python/Unit 5/Pairsumonious_Numbers.py
print("===== PAIRSUMONIOUS NUMBERS =====\n")

n = int(input("Enter number count: "))

print("\nEnter original numbers:")
print("Example: 1 2 3 4\n")

numbers = list(map(int, input().split()))

print("\nPair Sums:")

for i in range(n):

    for j in range(i + 1, n):

        print(
            numbers[i] + numbers[j],
            end=" "
        )